src/Domain/CoordinateMaps/TimeDependent/RotScaleTrans.cpp

```diff
@@ -281,94 +267,78 @@ std::optional<std::array<double, Dim>> RotScaleTrans<Dim>::inverse(
-        // b = 2\rho(\bar{\xi}}^2(R_{in} - R_{out}) - \rho R_{in}^2 E_{a}(t)),
-        // c = \rho^2(\bar{\xi}}^2 - (E_{a}(t) R_{in})^2)
-        // where \rho = R_{out}(E_{a}(t) - E_{b}(t)).
-        else {
-          double radial_scaling_factor =
-              outer_radius_ * (scale_a_of_t - scale_b_of_t);
-          double squared_radial_scaling_factor = square(radial_scaling_factor);
-          a = radius_squared * square(outer_radius_ - inner_radius_) -
-              squared_radial_scaling_factor * square(inner_radius_);
-          b = 2.0 * radial_scaling_factor *
-              (radius_squared * (inner_radius_ - outer_radius_) -
-               radial_scaling_factor * square(inner_radius_) * scale_a_of_t);
-          c = squared_radial_scaling_factor *
-              (radius_squared - (square(scale_a_of_t * inner_radius_)));
-          std::optional<std::array<double, 2>> roots = real_roots(a, b, c);
-          radial_scaling_factor = root_helper(roots);
-          for (size_t i = 0; i < Dim; i++) {
-            gsl::at(pre_rotation_result, i) /=
-                (scale_a_of_t + (1 - radial_scaling_factor));
-          }
-        }
+    ASSERT(scale_a_of_t != 0.0 and scale_b_of_t != 0.0,
+           "An expansion map "
+           "value was set to 0.0, this will cause an FPE. Expansion a: "
+               << scale_a_of_t << " expansion b: " << scale_b_of_t);
+    if (radius >= scale_b_of_t * outer_radius_) {
+      for (size_t i = 0; i < Dim; i++) {
+        gsl::at(pre_rotation_result, i) /= scale_b_of_t;
       }
-      // Rigid expansion
-    } else {
+    } else if (radius <= scale_a_of_t * inner_radius_) {
       for (size_t i = 0; i < Dim; i++) {
         gsl::at(pre_rotation_result, i) /= scale_a_of_t;
       }
+    } else {
+      // We need to solve a quadratic of the form w^2a + wb + c where
+      // a, b, and c change if you're closer to the inner radius. This
+      // is an optimization to reduce roundoff from SpEC.
+      double a =
+          square(scale_a_of_t * inner_radius_ - scale_b_of_t * outer_radius_);
+      double b = 0.;
+      double c = 0.;
+      double radial_scaling_factor = 0.;
+      double radius_squared = square(radius);
+      // When closer to the inner radius the quadratic has the form
+      // a = (E_{a}(t)R_{in} - E_{b}(t)R_{out})^2,
+      // b = 2E{b}(t)R_{out}(E_{a}(t)R_{in} - E_{b}(t)R_{out}),
+      // c = E_{b}(t)^2 R_{out}^2 - \bar{\xi}^2
+      if (1.0 - radius / (inner_radius_ * scale_a_of_t +
+                          outer_radius_ * scale_b_of_t) <
+          .5) {
+        radial_scaling_factor = scale_b_of_t * outer_radius_;
+        b = 2.0 * radial_scaling_factor *
+            (scale_a_of_t * inner_radius_ - radial_scaling_factor);
+        c = square(scale_b_of_t * outer_radius_) - radius_squared;
+        std::optional<std::array<double, 2>> roots = real_roots(a, b, c);
+        radial_scaling_factor = root_helper(roots);
+        radial_scaling_factor =
```

Comment:
You reuse `radial_scaling_factor` to represent different intermediate quantities. This can be somewhat confusing. Could you make some intermediate variable names for these instead and only set `raidal_scaling_factor` at the very end? Same for the `else` part of this `if`.